pelicanconf.py

```diff
@@ -61,8 +62,15 @@ def page_with_slug_index(pages):
         if page.slug == "index":
             return page
 
+def page_with_slug(pages, slug):
+    for page in pages:
+        if page.slug == slug:
+            return page
+
 JINJA_FILTERS = {
-    'page_with_slug_index': page_with_slug_index, }
+    'page_with_slug_index': page_with_slug_index,
```

Comment:
We can probably remove the `page_with_slug_index` entirely if we have a more generic one(?)